src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/OptionsFilter.php

```diff
@@ -0,0 +1,144 @@
+        array $supportedOperators = []
+    ) {
+        $this->attrValidatorHelper = $attributeValidatorHelper;
+        $this->supportedAttributeTypes = $supportedAttributeTypes;
+        $this->supportedOperators = $supportedOperators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addAttributeFilter(
+        AttributeInterface $attribute,
+        $operator,
+        $values,
+        $locale = null,
+        $scope = null,
+        $options = []
+    ) {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the filter.');
+        }
+
+        $this->checkLocaleAndChannel($attribute, $locale, $scope);
+
+        if (Operators::IS_EMPTY !== $operator && Operators::IS_NOT_EMPTY !== $operator) {
+            $this->checkValue($attribute, $values);
+        }
+
+        $attributePath = $this->getAttributePath($attribute, $locale, $scope);
+
+        switch ($operator) {
+            case Operators::IN_LIST:
+                $clause = [
+                    'terms' => [
+                        $attributePath => $values
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            case Operators::NOT_IN_LIST:
+                $notInListClause = [
+                    'terms' => [
+                        $attributePath => $values
+                    ]
+                ];
+
+                $existsClause = [
+                    'exists' => ['field' => $attributePath]
+                ];
+
+                $this->searchQueryBuilder->addMustNot($notInListClause);
+                $this->searchQueryBuilder->addFilter($existsClause);
+
+                break;
+            case Operators::IS_EMPTY:
+                $clause = [
+                    'exists' => ['field' => $attributePath]
+                ];
+
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => ['field' => $attributePath]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param AttributeInterface $attribute
+     * @param array              $values
+     *
+     * @throws ObjectNotFoundException
+     */
+    protected function checkValue(AttributeInterface $attribute, $values)
+    {
+        FieldFilterHelper::checkArray($attribute->getCode(), $values, static::class);
+
+        foreach ($values as $value) {
+            FieldFilterHelper::checkIdentifier($attribute->getCode(), $value, static::class);
+        }
+
+        $options = $attribute->getOptions();
+
+        $existingOptionCodes = [];
```

Comment:
do we do that in the original filter?
because here you load potentially a lot of objects